### PR TITLE
Updated sealed secret app to new style

### PR DESCRIPTION
**Pull request by nitishkumar71**
## Description

1. Updated the code in sealed_secret_app.go
## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))
#256 
#220 

## How Has This Been Tested?

1. Install sealed secret into cluster `arakade install sealed-secrets`
2. Install client-side tool into /usr/local/bin/
```
GOOS=$(go env GOOS)
GOARCH=$(go env GOARCH)
wget https://github.com/bitnami-labs/sealed-secrets/releases/download/v0.12.4/kubeseal-$GOOS-$GOARCH
sudo install -m 755 kubeseal-$GOOS-$GOARCH /usr/local/bin/kubeseal
```
3. Create a sealed secret file
```
kubectl create secret generic secret-name --dry-run --from-literal=foo=bar -o [json|yaml] | \
 kubeseal \
 --controller-name=sealed-secrets \
 --controller-namespace=default \
 --format [json|yaml] > mysealedsecret.[json|yaml]
```
4. Apply the sealed secret
```
kubectl create -f mysealedsecret.[json|yaml]
```
5. Decrypt secret
```
kubectl get secret secret-name -o [json|yaml]
```
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

- [ ] I have tested this on arm, or have added code to prevent deployment

### CODE REVIEW

**Review by Waterdrips — APPROVED** [review]
LGTM

**Comment by alexellis** [comment]
Sorry this will require a rebase due to #258 being merged.

@Waterdrips what does @nitishkumar71 need to do to rebase this?

**Comment by Waterdrips** [comment]
yep, delete the section at the top with `kubeconfig` and change it to 

```go 
kubeConfigPath, _ := command.Flags().GetString("kubeconfig")
```

Then call options.WithKubeconfigPath(kubeConfigPath) into the installer options section.

This will fix the issue with not using a supplied kubeconfig flag

**Comment by nitishkumar71** [comment]
> yep, delete the section at the top with `kubeconfig` and change it to
> 
> ```go
> kubeConfigPath, _ := command.Flags().GetString("kubeconfig")
> ```
> 
> Then call options.WithKubeconfigPath(kubeConfigPath) into the installer options section.
> 
> This will fix the issue with not using a supplied kubeconfig flag

@Waterdrips I have made the changes. please review the changes.

File: cmd/apps/sealed_secret_app.go [hunk 1]
```diff
@@ -36,7 +38,6 @@ func MakeInstallSealedSecrets() *cobra.Command {
 		if err := config.SetKubeconfig(kubeConfigPath); err != nil {
```
Comment (Waterdrips) [inline cmd/apps/sealed_secret_app.go:38]: we dont need to set it here, its gets set in the chart app

**Review by Waterdrips — APPROVED** [review]

**Review by Waterdrips — CHANGES_REQUESTED** [review]
see comment on last review, no need to set kubeconfig -its done in the chart app

File: cmd/apps/sealed_secret_app.go (hunk 1 above)
Comment (nitishkumar71) [reply cmd/apps/sealed_secret_app.go]: Thanks for mention. I have made the changes. @Waterdrips

**Comment by alexellis** [comment]
@Waterdrips ready for another review.

File: cmd/apps/sealed_secret_app.go [hunk 2]
```diff
@@ -90,11 +73,26 @@ func MakeInstallSealedSecrets() *cobra.Command {
 			return err
 		}
 
-		err = helm.Helm3Upgrade("stable/sealed-secrets",
-			namespace, "values.yaml", defaultVersion, overrides, wait)
+		sealedSecretAppOptions := types.DefaultInstallOptions().
+			WithNamespace(namespace).
+			WithHelmPath(path.Join(userPath, ".helm")).
+			WithHelmRepo("stable/sealed-secrets").
+			WithHelmURL("https://charts.helm.sh/stable").
```
Comment (alexellis) [inline cmd/apps/sealed_secret_app.go:80]: Bitnami should have a new home for the chart. Can you look around and see if you can find it? The "stable helm repo" is going to be deprecated shortly.

**Comment by alexellis** [comment]
FYI https://twitter.com/HelmPack/status/1322224008822202368?s=20

I've also checked the Bitnami Charts repo (nothing there yet) and asked on their `#sealed-secrets` channel on K8s Slack.

**Comment by alexellis** [comment]
I found out here -> https://github.com/bitnami-labs/sealed-secrets/issues/389 

Thanks @mkmik

**Comment by nitishkumar71** [comment]
> I found out here -> [bitnami-labs/sealed-secrets#389](https://github.com/bitnami-labs/sealed-secrets/issues/389)
> 
> Thanks @mkmik

@alexellis Thanks for pointing it out. I think, we should wait for the new chart repo of sealed secrets. What do you suggest?

**Comment by alexellis** [comment]
No, just go ahead. We already have this pointing at the stable repo.

**Review by alexellis — APPROVED** [review]
Approved

**Comment by alexellis** [comment]
In retrospect and due to #274 I think we do need to switch over to the new helm repo, but it should be a one line change if you can work on that today @nitishkumar71 ?

My understanding is that the charts have now moved to `https://charts.helm.sh/stable` as read-only.

**Pull request merged**